main/lsp/requests/workspace_symbols.cc

```diff
@@ -21,12 +44,134 @@ unique_ptr<SymbolInformation> symbolRef2SymbolInformation(const LSPConfiguration
     if (location == nullptr) {
         return nullptr;
     }
-    auto result =
-        make_unique<SymbolInformation>(sym->name.show(gs), symbolRef2SymbolKind(gs, symRef), std::move(location));
+    auto result = new SymbolInformation(sym->name.show(gs), symbolRef2SymbolKind(gs, symRef), std::move(location));
```

Comment:
!!!!

This looks like a memory leak. You're dynamically allocating `result` w/ `new`, but never `delete`ing it.

You should return `unique_ptr<SymbolInformation>`, which can be `nullptr` AND will automatically clean up memory (call `delete`) once that value is destructed.

EDIT: I see that this gets coerced into a unique_ptr later with `emplace_back`, but I'd vastly prefer you create a unique_ptr here and `move` it into the array.

Reply:
(We don't use `new` anywhere in the codebase, AFAIK)

Reply:
Thanks for setting me straight, that totally makes sense!  Fixing now...